Refactor few code style rules to separate docs (#20856)
dotnet/docs: docs/fundamentals/code-analysis/style-rules/index.md

   Rules that pertain to the naming of code elements. For example, you can specify that `async` method names must have an "Async" suffix.

## Index

> The following table list all the code style rules by ID and option(s), if any.

Reviewer: ```suggestion
The following table list all the code style rules by ID and options, if any.
```

> The following pages list all the code style rules by rule ID and corresponding option(s), if any.

Author: Rule ID + option based index.

> The following pages list all the code style rules by rule ID and corresponding option(s), if any.

Author: ```suggestion
The following table list all the code style rules by ID and option(s), if any.
```

> [!div class="mx-tdCol4BreakAll"]

> > | Rule ID | Title | Option | Option Group |

Reviewer: ```suggestion
> | Rule ID | Title | Option | Option group |
```

> | - | - | - | - |

> > | [IDE0003](ide0003_ide0009.md) | Remove 'this' or 'Me' qualification | [dotnet_style_qualification_for_field](ide0003_ide0009.md#dotnet\_style\_qualification\_for_field)<br/> [dotnet_style_qualification_for_property](ide0003_ide0009.md#dotnet\_style\_qualification\_for_property)<br/> [dotnet_style_qualification_for_method](ide0003_ide0009.md#dotnet\_style\_qualification\_for_method)<br/> [dotnet_style_qualification_for_event](ide0003_ide0009.md#dotnet\_style\_qualification\_for_event) | 'this.' and 'Me.' qualifiers

Reviewer: ```suggestion
> | [IDE0003](ide0003_ide0009.md) | Remove 'this' or 'Me' qualification | [dotnet_style_qualification_for_field](ide0003_ide0009.md#dotnet\_style\_qualification\_for_field)<br/> [dotnet_style_qualification_for_property](ide0003_ide0009.md#dotnet\_style\_qualification\_for_property)<br/> [dotnet_style_qualification_for_method](ide0003_ide0009.md#dotnet\_style\_qualification\_for_method)<br/> [dotnet_style_qualification_for_event](ide0003_ide0009.md#dotnet\_style\_qualification\_for_event) | 'this.' and 'Me.' qualifiers |
```

> > | [IDE0009](ide0003_ide0009.md) | Add 'this' or 'Me' qualification | [dotnet_style_qualification_for_field](ide0003_ide0009.md#dotnet\_style\_qualification\_for_field)<br/> [dotnet_style_qualification_for_property](ide0003_ide0009.md#dotnet\_style\_qualification\_for_property)<br/> [dotnet_style_qualification_for_method](ide0003_ide0009.md#dotnet\_style\_qualification\_for_method)<br/> [dotnet_style_qualification_for_event](ide0003_ide0009.md#dotnet\_style\_qualification\_for_event) | 'this.' and 'Me.' qualifiers

Reviewer: ```suggestion
> | [IDE0009](ide0003_ide0009.md) | Add 'this' or 'Me' qualification | [dotnet_style_qualification_for_field](ide0003_ide0009.md#dotnet\_style\_qualification\_for_field)<br/> [dotnet_style_qualification_for_property](ide0003_ide0009.md#dotnet\_style\_qualification\_for_property)<br/> [dotnet_style_qualification_for_method](ide0003_ide0009.md#dotnet\_style\_qualification\_for_method)<br/> [dotnet_style_qualification_for_event](ide0003_ide0009.md#dotnet\_style\_qualification\_for_event) | 'this.' and 'Me.' qualifiers |
```

> > | [IDE0049](ide0049.md) | Use language keywords instead of framework type names for type references | [dotnet_style_predefined_type_for_locals_parameters_members](ide0049.md#dotnet\_style\_predefined\_type\_for\_locals\_parameters_members)<br/> [dotnet_style_predefined_type_for_member_access](ide0049.md#dotnet\_style\_predefined\_type\_for\_member_access)<br/> | Use language keywords instead of framework type names for type references

Reviewer: ```suggestion
> | [IDE0049](ide0049.md) | Use language keywords instead of framework type names for type references | [dotnet_style_predefined_type_for_locals_parameters_members](ide0049.md#dotnet\_style\_predefined\_type\_for\_locals\_parameters_members)<br/> [dotnet_style_predefined_type_for_member_access](ide0049.md#dotnet\_style\_predefined\_type\_for\_member_access)<br/> | Use language keywords instead of framework type names for type references |
```

   Rules that pertain to the naming of code elements. For example, you can specify that `async` method names must have an "Async" suffix.

## Index

The following table list all the code style rules by ID and options, if any.

> [!div class="mx-tdCol4BreakAll"]
> | Rule ID | Title | Option | Option group |
> | - | - | - | - |

> > | [IDE0003](ide0003-ide0009.md) | Remove 'this' or 'Me' qualification | [dotnet_style_qualification_for_field](ide0003-ide0009.md#dotnet\_style\_qualification\_for_field)<br/> [dotnet_style_qualification_for_property](ide0003-ide0009.md#dotnet\_style\_qualification\_for_property)<br/> [dotnet_style_qualification_for_method](ide0003-ide0009.md#dotnet\_style\_qualification\_for_method)<br/> [dotnet_style_qualification_for_event](ide0003-ide0009.md#dotnet\_style\_qualification\_for_event) | 'this.' and 'Me.' preferences |

Reviewer: This table is hard to read because it's getting a horizontal scroll bar. I think it might be better to have a separate table for each option group, that way the tables only have 3 columns instead of 4.

Author: The issue with that is IDs are not used consecutively, so there is no single place where one could see a single table with a sorted list of all IDs.

Author: We can also consider removing the Option group column from this table. Would that be better?

I also had a question on use of  `\_` instead of `_` in the option headers. It seems that using `_` everywhere does not impact any rendering. Will it be okay for me to make this change everywhere in the refactored pages? Makes it much cleaner to read and edit the markdown...

Reviewer: Yes and yes!
